rs/monitoring/metrics/src/adapter_metrics_registry.rs

```diff
@@ -56,22 +55,21 @@ impl AdapterMetricsRegistry {
     pub fn register(&self, adapter_metrics: AdapterMetrics) -> Result<(), Error> {
         if self
             .adapters
-            .blocking_read()
+            .read()
             .iter()
             .any(|a| a.get_name() == adapter_metrics.get_name())
         {
             return Err(Error::AlreadyReg);
         }
-        self.adapters.blocking_write().push(adapter_metrics);
+        self.adapters.write().push(adapter_metrics);
         Ok(())
```

Comment:
Given that the error case should rarely happen, can't we avoid the read lock and hold the write lock for longer?